### ISSUE TITLE
Add support for MySQL FULLTEXT KEY inline declaration in CREATE TABLE

### ISSUE THREAD

**Issue by rsmckinney** [jOOQ/jOOQ#9161]
### Expected behavior and actual behavior:

MySql Create Table syntax fails to parse. This is crucial for a DDL parser.

### Steps to reproduce the problem (if possible, create an MCVE: https://github.com/jOOQ/jOOQ-mcve):

- Download and extract the standard [sakila-schema](https://downloads.mysql.com/docs/sakila-db.zip) from MySql.
- Parse the `sakila-schema.sql` file:
```java
...
DSLContext ctx = DSL.using(SQLDialect.MYSQL);
Meta meta = ctx.meta(Source.of(sakila));
```
- Results in ParserException:
```
Information:java: org.jooq.impl.ParserException: Token ')' expected: [180:16] ...on TEXT,
Information:java:   PRIMARY KEY  (film_id),
Information:java:   FULLTEXT KEY [*]idx_title_description (title,description)
Information:java: )ENGINE=InnoDB DEFAULT CHARSET=utf8;
Information:java: ...
Information:java: 	at org.jooq.impl.ParserContext.expected(ParserImpl.java:10641)
Information:java: 	at org.jooq.impl.ParserImpl.parse(ParserImpl.java:10120)
Information:java: 	at org.jooq.impl.ParserImpl.parse(ParserImpl.java:10115)
Information:java: 	at org.jooq.impl.ParserImpl.parseCreateTable(ParserImpl.java:3370)
Information:java: 	at org.jooq.impl.ParserImpl.parseCreate(ParserImpl.java:2143)
Information:java: 	at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:844)
Information:java: 	at org.jooq.impl.ParserImpl.parse(ParserImpl.java:542)
Information:java: 	at org.jooq.impl.ParserImpl.parse(ParserImpl.java:529)
Information:java: 	at org.jooq.impl.DDLDatabaseInitializer.loadScript(DDLDatabaseInitializer.java:172)
Information:java: 	at org.jooq.impl.DDLDatabaseInitializer.initializeUsing(DDLDatabaseInitializer.java:153)
Information:java: 	at org.jooq.impl.DDLMetaProvider.provide(DDLMetaProvider.java:69)
Information:java: 	at org.jooq.impl.DefaultDSLContext.meta(DefaultDSLContext.java:443)
Information:java: 	at manifold.sql.type.SqlModel.parse(SqlModel.java:94)
```

### Versions:

- jOOQ: 3.12.0
- Java: 8
- JDBC Driver (include name if inofficial driver): H2

**Comment by lukaseder** [comment]
Thank you very much for your report. This should definitely parse, we recognise this syntax. Will investigate.

**Comment by lukaseder** [comment]
My mistake. We support these syntaxes:

```sql
create fulltext key a on t(x)
alter table t add fulltext key a (x)
```

But not the one used in the Sakila schema. We'll add support for it right away (and other syntax used in the Sakila database definition)

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.13.0. Backport scheduled for 3.12.2 (#9173).

Thanks again!
